ui/server/src/main/scala/pl/touk/nussknacker/ui/api/WebResources.scala

```diff
@@ -1,36 +1,73 @@
 package pl.touk.nussknacker.ui.api
 
-import java.nio.file.Files
-import akka.http.scaladsl.server.{Directives, Route}
+import akka.http.scaladsl.server.{Directive1, Directives, Route}
 import com.typesafe.scalalogging.LazyLogging
 import org.apache.commons.io.{FileUtils, IOUtils}
+import pl.touk.nussknacker.engine.util.cache.{CacheConfig, DefaultCache}
+import pl.touk.nussknacker.ui.NotFoundError
+
+import java.io.File
+import java.net.URL
+import java.nio.file.Files
 
 class WebResources(publicPath: String) extends Directives with LazyLogging {
 
-  //see config.js comment
-  private lazy val mainContentFile = {
-    val tempMainContentFile = Files.createTempFile("nussknacker", "main.html").toFile
-    tempMainContentFile.deleteOnExit()
-    val data = getClass.getClassLoader.getResourceAsStream("web/static/main.html")
-    val content = Option(data).map(IOUtils.toString).getOrElse {
-      logger.error("Failed to find web/static/main.html - probably frontend resources are not packaged in jar. Frontend won't work properly!")
-      ""
-    }
-    FileUtils.writeStringToFile(tempMainContentFile, content.replace("__publicPath__", publicPath))
-    tempMainContentFile
-  }
+  private val mainFilesCache = new DefaultCache[String, File](CacheConfig())
+
+  private val defaultMainFilePath = "web/static/main.html"
 
   val route: Route =
-    pathPrefix("static") {
+    pathPrefix("submodules" / Segment /) { moduleName =>
+      (get & submodule(moduleName)) { _ =>
+        extractRequest { matched =>
```

Comment:
Why can't we use getFromResourceDirectory? Do we really need index.html in submodules?

Reply:
I've changed it, right now we don't wan't so complex routing..